components/repeater/index.js

```diff
@@ -0,0 +1,82 @@
+import { useBlockEditContext } from '@wordpress/block-editor';
+import { useSelect, dispatch } from '@wordpress/data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+/**
+ * The Repeater Component.
+ *
+ * @param {object} props React props
+ * @param {Function} props.children Render prop to render the children.
+ * @param {string} props.attribute property of the block attribute that will provide data for Repeater.
+ * @returns {*} React JSX
+ */
+export const Repeater = ({ children, attribute }) => {
+	const { clientId, name } = useBlockEditContext();
+	const { updateBlockAttributes } = dispatch('core/editor');
+
+	const { repeaterData, defaultRepeaterData } = useSelect((select) => {
+		const { getBlockAttributes } = select('core/editor');
+		const { getBlockType } = select('core/blocks');
+
+		return {
+			repeaterData: getBlockAttributes(clientId),
+			defaultRepeaterData: getBlockType(name).attributes[attribute].default,
+		};
+	});
+
+	/**
+	 * Adds a new repeater item.
+	 */
+	function addItem() {
+		updateBlockAttributes(clientId, {
+			[attribute]: [...repeaterData[attribute], defaultRepeaterData],
+		});
+	}
+
+	/**
+	 * Updates the item currently being edited.
+	 *
+	 * @param {number} index The index at which the item should be updated.
+	 * @param {string|number|boolean} value The value that should be used to updated the item.
+	 */
+	function setItem(index, value) {
```

Comment:
I'd love it if the setitem function and all the others already has the index inherited from the map and therefore the developer doesn't need to set it manually anymore :)

Reply:
I should've done this! Added in [debedc6](https://github.com/10up/block-components/pull/110/commits/debedc678c80857e9d4de9342eaa2f8522db8e84)
Thank you!